dynamic-config/config-cc/src/main/java/org/apache/servicecomb/config/client/ParseConfigUtils.java

```diff
@@ -37,35 +39,63 @@
   private static final Logger LOGGER = LoggerFactory.getLogger(ParseConfigUtils.class);
 
   private static LinkedHashMap<String, Map<String, Object>> multiDimensionItems = new LinkedHashMap<>();
+  //it's dangerous that makes flatItems public
+  private static final Map<String, Object> flatItems = new HashMap<>();
 
-  public static final Map<String, Object> flatItems = new HashMap<>();
+  private static String currentVersionInfo = "default";
 
   private UpdateHandler updateHandler;
 
+  private Lock configLock = new ReentrantLock();
+
   public ParseConfigUtils(UpdateHandler updateHandler) {
     this.updateHandler = updateHandler;
   }
 
+  /*
+    as the data is returned, we can block the thread at a short time. consider that if the multiple verticle is deployed
+    and if we use pull mode and push mode at the same time , we must share a common lock with all methods which would
+    change the config setting
+   */
   public void refreshConfigItems(Map<String, Map<String, Object>> remoteItems) {
-    multiDimensionItems.clear();
-    multiDimensionItems.putAll(remoteItems);
-    doRefreshItems();
-    LOGGER.debug("refresh config success");
+    try {
+      configLock.lock();
```

Comment:
This concurrent control may not work as expected. Because data holding in this class is static, and lock is instance level. 
Anyway, this class only have one instance in runtime in this module, it will cause no problem. It's better we can modify this.